python/ffsim/states/slater.py

```diff
@@ -0,0 +1,353 @@
+from __future__ import annotations
+
+import numpy as np
+import scipy.linalg
+
+
+def sample_slater(
+    rdm: np.ndarray | tuple[np.ndarray, np.ndarray],
+    chain_length: int,
+    n_chains: int,
+    n_particles_to_move: int = 1,
+    seed=None,
+) -> np.ndarray:
```

Comment:
For now, let's make the input and output formats match [`ffsim.sample_statevector`](https://qiskit-community.github.io/ffsim/api/ffsim.html#ffsim.sample_state_vector), which has this signature:
```python
def sample_state_vector(
    rdm: np.ndarray | tuple[np.ndarray, np.ndarray],
    *,
    orbs: list[int],
    shots: int,
    norb: int | None = None,
    nelec: int | tuple[int, int] | None = None,
    seed: np.random.Generator | int | None = None,
) -> list[str]:
```
So here, we would have
```python
def sample_slater(
    vec: np.ndarray | StateVector,
    *,
    orbs: list[int],
    shots: int,
    norb: int | None = None,
    nelec: int | tuple[int, int] | None = None,
    seed: np.random.Generator | int | None = None,
) -> list[str]:
```
For the other parameters you have, like `chain_length`, you can add those as extra parameters, but with some reasonable default values.

Note that the return type is a list of strings of length `shots`. This isn't the most efficient, but we can revisit this in the future if needed. Or, if you think it's important to allow the return type to be a bit array, perhaps we can handle that as an additional option, which we should probably define an enum for. Something like
```python
return_type: BitstringType = BitstringType.STRING   # enum representing either string, bit array, or integer
```